MetaMask/metamask-extension
# Replace deprecated Box component

## Explanation

Replaces deprecated Box component with updated version.

* Fixes #20160
* Fixes #20159
* Fixes #20155

## Screenshots/Screencaps

# Before
![image](https://github.com/MetaMask/metamask-extension/assets/8605618/89dbf2b7-05d6-4eaf-a5a1-3d27a8c46217)

# After

![image](https://github.com/MetaMask/metamask-extension/assets/8605618/bd97b501-a675-4dcc-870a-db53e9e070a5)

## Manual Testing Steps

Check that the updated components are visually unaffected, e.g. the token list, and signature request, add network and token allowance modals.

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
